ovirt_imageio/client/_options.py

```diff
@@ -280,3 +300,45 @@ def __call__(self, s):
             raise ValueError(f"Size {s!r} > {self.maximum}")
 
         return value
+
+
+class Type:
+    """
+    Wrapper to expose function validators as types.
+    """
+
+    def __init__(self, name, func):
+        self.name = name
+        self.func = func
+
+    def __call__(self, value):
+        return self.func(value)
+
+    def __repr__(self):
+        return self.name
+
+
+def _validate_uuid(id):
+    try:
+        return str(uuid.UUID(id))
+    except ValueError as e:
+        raise ValueError(f"Invalid UUID value: '{id}': {str(e)}") from e
+
+
+def _validate_file(filename):
+    """
+    Validate that the file exists.
+
+    Raises:
+        ValueError
+
+    Returns:
+        str: filename
+    """
+    if not os.path.exists(filename):
```

Comment:
But this will fail with "path does not exist" which is wrong. Best do:

    if not exist:
        fail with "does not exist"
    if not file:
        fail with "not a file" error